<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1230</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Scale-aware Black-and-White Abstraction of 3D Shapes</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab_basic">
       <attribute name="title">
        <string>ModelViewer</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="ModelViewer" name="modelViewer"/>
        </item>
        <item>
         <widget class="QWidget" name="widget_2" native="true">
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QGroupBox" name="groupBox_4">
             <property name="title">
              <string>Display</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <layout class="QFormLayout" name="formLayout_9">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_35">
                  <property name="text">
                   <string>Triangle       </string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QCheckBox" name="mv_display_triangle">
                  <property name="text">
                   <string/>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_36">
                  <property name="text">
                   <string>Edge</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QCheckBox" name="mv_display_edge">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_37">
                  <property name="text">
                   <string>Vertex</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QCheckBox" name="mv_display_vertex">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_7">
             <property name="title">
              <string>Directional Light</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <layout class="QFormLayout" name="formLayout_5">
                <item row="0" column="0">
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>Phi</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QSlider" name="horizontalSlider">
                  <property name="maximum">
                   <number>359</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="tickPosition">
                   <enum>QSlider::TicksAbove</enum>
                  </property>
                  <property name="tickInterval">
                   <number>50</number>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_2">
                  <property name="text">
                   <string>Theta</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QSlider" name="horizontalSlider_2">
                  <property name="maximum">
                   <number>359</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="tickPosition">
                   <enum>QSlider::TicksAbove</enum>
                  </property>
                  <property name="tickInterval">
                   <number>50</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_9">
             <property name="title">
              <string>Mesh Information</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_13">
              <item>
               <layout class="QFormLayout" name="formLayout_6">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_26">
                  <property name="text">
                   <string>Vertices</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QLabel" name="mvNVertices">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_28">
                  <property name="text">
                   <string>Faces</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLabel" name="mvNFaces">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_segment">
       <attribute name="title">
        <string>BWAbstraction</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <item>
         <widget class="BWARenderer" name="segmentBasedRenderer"/>
        </item>
        <item>
         <widget class="QWidget" name="widget_4" native="true">
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_11">
           <property name="rightMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QGroupBox" name="groupBox_17">
             <property name="title">
              <string>Background</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_26">
              <item>
               <layout class="QFormLayout" name="formLayout">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_7">
                  <property name="text">
                   <string>Color            </string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QComboBox" name="bwaBackgroundColor">
                  <item>
                   <property name="text">
                    <string>White</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Grey</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Black</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_5">
             <property name="title">
              <string>Parameters</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <layout class="QFormLayout" name="formLayout_3">
                <item row="2" column="0">
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>Scale</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QDoubleSpinBox" name="bwaScaleFactor">
                  <property name="minimum">
                   <double>0.000000000000000</double>
                  </property>
                  <property name="maximum">
                   <double>1.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.050000000000000</double>
                  </property>
                  <property name="value">
                   <double>1.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="label_11">
                  <property name="text">
                   <string>Background W.</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QDoubleSpinBox" name="bwaBackgroundWeight">
                  <property name="decimals">
                   <number>3</number>
                  </property>
                  <property name="maximum">
                   <double>100000.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>10.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>250.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="4" column="0">
                 <widget class="QLabel" name="label_10">
                  <property name="text">
                   <string>Consistency W.</string>
                  </property>
                 </widget>
                </item>
                <item row="4" column="1">
                 <widget class="QDoubleSpinBox" name="bwaConsistencyWeight">
                  <property name="decimals">
                   <number>3</number>
                  </property>
                  <property name="maximum">
                   <double>100000.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>10.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>50.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="5" column="0">
                 <widget class="QLabel" name="label_12">
                  <property name="text">
                   <string>Neighbour W.</string>
                  </property>
                 </widget>
                </item>
                <item row="5" column="1">
                 <widget class="QDoubleSpinBox" name="bwaNeighbourWeight">
                  <property name="decimals">
                   <number>3</number>
                  </property>
                  <property name="maximum">
                   <double>100000.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>10.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>50.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="6" column="0">
                 <widget class="QLabel" name="label_9">
                  <property name="text">
                   <string>Inclusion W.</string>
                  </property>
                 </widget>
                </item>
                <item row="6" column="1">
                 <widget class="QDoubleSpinBox" name="bwaInclusionWeight">
                  <property name="decimals">
                   <number>3</number>
                  </property>
                  <property name="maximum">
                   <double>100000.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>1.500000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="7" column="0">
                 <widget class="QLabel" name="label_16">
                  <property name="text">
                   <string>Contrast W. (%)</string>
                  </property>
                 </widget>
                </item>
                <item row="7" column="1">
                 <widget class="QDoubleSpinBox" name="bwaContrastWeight">
                  <property name="decimals">
                   <number>3</number>
                  </property>
                  <property name="maximum">
                   <double>100.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>Render Width</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QSpinBox" name="bwaRenderWidth">
                  <property name="minimum">
                   <number>40</number>
                  </property>
                  <property name="maximum">
                   <number>10000</number>
                  </property>
                  <property name="singleStep">
                   <number>100</number>
                  </property>
                  <property name="value">
                   <number>1600</number>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_4">
                  <property name="text">
                   <string>Render Height</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QSpinBox" name="bwaRenderHeight">
                  <property name="minimum">
                   <number>30</number>
                  </property>
                  <property name="maximum">
                   <number>10000</number>
                  </property>
                  <property name="singleStep">
                   <number>100</number>
                  </property>
                  <property name="value">
                   <number>1200</number>
                  </property>
                 </widget>
                </item>
                <item row="8" column="0">
                 <widget class="QLabel" name="label_18">
                  <property name="text">
                   <string>Feature W. (%)</string>
                  </property>
                 </widget>
                </item>
                <item row="8" column="1">
                 <widget class="QDoubleSpinBox" name="bwaFeatureLineThresholdPercent">
                  <property name="decimals">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <double>10000.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>100.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>1000.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_13">
             <property name="title">
              <string>Result Display</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_21">
              <item>
               <layout class="QFormLayout" name="formLayout_11">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_39">
                  <property name="text">
                   <string>Display</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QComboBox" name="bwaResultDisplay">
                  <property name="currentText">
                   <string>BWA</string>
                  </property>
                  <item>
                   <property name="text">
                    <string>BWA</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Patch</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Depth Critical Line</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Boundary</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Sharp Edge Line</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Feature Line</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Consistency</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Component</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Inclusion</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Distance Transform</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_4">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1230</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionImportMesh"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuScreen">
    <property name="title">
     <string>Screen</string>
    </property>
    <addaction name="actionSave_Screen"/>
   </widget>
   <widget class="QMenu" name="menuCamera">
    <property name="title">
     <string>Camera</string>
    </property>
    <addaction name="actionToggle_Ortho_Projection"/>
    <addaction name="actionReset_Camera"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuScreen"/>
   <addaction name="menuCamera"/>
  </widget>
  <action name="actionLoad">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionLog">
   <property name="text">
    <string>Show Application Log...</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Reset View</string>
   </property>
  </action>
  <action name="actionToggle_Ortho">
   <property name="text">
    <string>Toggle Ortho</string>
   </property>
  </action>
  <action name="actionSave_Screen">
   <property name="text">
    <string>Save Screen</string>
   </property>
  </action>
  <action name="actionImport_Mesh">
   <property name="text">
    <string>Import Mesh...</string>
   </property>
  </action>
  <action name="actionToggle_Ortho_Projection">
   <property name="text">
    <string>Toggle Ortho Projection</string>
   </property>
  </action>
  <action name="actionReset_Camera">
   <property name="text">
    <string>Reset Camera</string>
   </property>
  </action>
  <action name="actionSave_Camera_Config">
   <property name="text">
    <string>Save Camera Config</string>
   </property>
  </action>
  <action name="actionLoad_Camera_Config">
   <property name="text">
    <string>Load Camera Config</string>
   </property>
  </action>
  <action name="actionReimport_All">
   <property name="text">
    <string>Reimport All</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionImportMesh">
   <property name="text">
    <string>Load Model...</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="actionBatch_Render_All">
   <property name="text">
    <string>Batch Render All</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ModelViewer</class>
   <extends>QOpenGLWidget</extends>
   <header>renderer/modelviewer.h</header>
  </customwidget>
  <customwidget>
   <class>BWARenderer</class>
   <extends>QOpenGLWidget</extends>
   <header>renderer/bwarenderer.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
